osu.Game/Users/UserStatistics.cs

```diff
@@ -26,18 +26,20 @@ public struct LevelInfo
             public int Progress;
         }
 
-        [JsonProperty(@"pp")]
-        public decimal? PP;
-
-        [JsonProperty(@"pp_rank")] // the API sometimes only returns this value in condensed user responses
-        private int? rank
-        {
-            set => Ranks.Global = value;
-        }
+        [JsonProperty(@"global_rank")]
+        public int? GlobalRank;
```

Comment:
So there are three properties now for ranks? `GlobalRank`, `Ranks` and `PPRank`? Which one is valid when? Why do none of those have xmldocs (or obsoletion notices, if they apply) at all? This model design borders on intentional obfuscation.

Reply:
`GlobalRank` is what will always stay there, `Ranks.Global` is what's likely to be nuked in favour of `GlobalRank`, `PPRank` is still in discussion whether to remove it, or keep it and rename it to better descriptive name.

Reply:
Actually, `Ranks.Global` is already removed, so you can scratch that second one. Could make `Ranks` private and expose a `CountryRank` field as well, if that's good for you?

Reply:
I've went ahead and done it, should read better now.